osu.Game/Skinning/LocalSkinOverrideContainer.cs

```diff
@@ -43,22 +43,30 @@ public SampleChannel GetSample(string sampleName)
             return fallbackSource?.GetSample(sampleName);
         }
 
-        public TValue? GetValue<TConfiguration, TValue>(Func<TConfiguration, TValue?> query) where TConfiguration : SkinConfiguration where TValue : struct
+        public TValue GetValue<TConfiguration, TValue>(Func<TConfiguration, TValue> query) where TConfiguration : SkinConfiguration
         {
-            TValue? val = null;
+            TValue val;
             if ((source as Skin)?.Configuration is TConfiguration conf)
-                val = query?.Invoke(conf);
+                if (beatmapSkins && (val = query.Invoke(conf)) != null)
+                    return val;
 
-            return val ?? fallbackSource?.GetValue(query);
+            return fallbackSource == null ? default : fallbackSource.GetValue(query);
         }
 
-        public TValue GetValue<TConfiguration, TValue>(Func<TConfiguration, TValue> query) where TConfiguration : SkinConfiguration where TValue : class
+        public bool TryGetValue<TConfiguration, TValue>(Func<TConfiguration, TValue, bool> query, out TValue val) where TConfiguration : SkinConfiguration
```

Comment:
What does the `TValue` in the `Func` do? Seems unnecessary?

Reply:
removed TryGetValue for now as per discussion (untested and unlikely to work).